src/ansys/geometry/core/geometry/box_uv.py

```diff
@@ -0,0 +1,184 @@
+    TopCenter = 2
+    TopRight = 3
+    BottomLeft = 4
+    BottomCenter = 5
+    BottomRight = 6
+    LeftCenter = 7
+    RightCenter = 8
+    Center = 9
+
+
+class BoxUV:
+    """BoxUV class."""
+
+    def __init__(self, range_u: Interval = None, range_v: Interval = None) -> None:
+        """Root constructor for BoxUV."""
+        if range_u is not None:
+            self._interval_u = range_u
+        if range_v is not None:
+            self._interval_v = range_v
+
+    @classmethod
+    def from_param(cls, param: ParamUV):
+        """Secondary constructor for BoxUV using a ParamUV object type."""
+        return cls(Interval(param.u, param.u), Interval(param.v, param.v))
+
+    @classmethod
+    def from_two_params(cls, param1: ParamUV, param2: ParamUV):
+        """Secondary constructor for BoxUV using two ParamUV object types."""
+        return cls(
+            Interval(min(param1.u, param2.u), max(param1.u, param2.u)),
+            Interval(min(param1.v, param2.v), max(param1.v, param2.v)),
+        )
+
+    @property
+    def interval_u(self) -> Interval:
+        """Return the u interval."""
+        return self._interval_u
+
+    @property
+    def interval_v(self) -> Interval:
+        """Return the v interval."""
+        return self._interval_v
+
+    def __eq__(self, other: object) -> bool:
+        """Check whether two BoxUV instances are equal."""
+        if not isinstance(other, BoxUV):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+        return self.interval_u.__eq__(other.interval_u) and self.interval_v.__eq__(other.interval_v)
+
+    def __ne__(self, other: object) -> bool:
+        """Check whether two BoxUV instances are not equal."""
+        if not isinstance(other, BoxUV):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+        return not self.interval_u.__eq__(other.interval_u) or not self.interval_v.__eq__(
+            other.interval_v
+        )
+
+    def is_empty(self):
+        """Return whether this BoxUV is empty."""
+        return self.interval_u.is_empty() or self.interval_v.is_empty()
+
+    def proportion(self, prop_u: Real, prop_v: Real) -> ParamUV:
+        """Evaluate the BoxUV at the given proportions."""
+        return ParamUV(
+            self.interval_u.get_relative_val(prop_u), self.interval_v.get_relative_val(prop_v)
+        )
+
+    def get_center(self) -> ParamUV:
+        """Evaluate the BoxUV in the middle."""
+        return self.proportion(0.5, 0.5)
+
+    def is_negative(self, tolerance_u: Real, tolerance_v: Real) -> bool:
+        """Check whether the BoxUV is negative."""
+        if self.is_empty():
+            return False
+        return self.interval_u.is_negative(tolerance_u) or self.interval_v.is_negative(tolerance_v)
+
+    @staticmethod
+    def unite(first: "BoxUV", second: "BoxUV") -> "BoxUV":
+        """Return the union of two BoxUV instances."""
```

Comment:
```suggestion
        """Unite of two BoxUV instances."""
```

Shouldn't there be ``Parameters" sections for this method and others?